src/wallet/wallet.cpp

```diff
@@ -2169,14 +2169,21 @@ bool CWallet::GetNewChangeDestination(const OutputType type, CTxDestination& des
     return true;
 }
 
-int64_t CWallet::GetOldestKeyPoolTime() const
+std::optional<int64_t> CWallet::GetOldestKeyPoolTime() const
 {
     LOCK(cs_wallet);
-    int64_t oldestKey = std::numeric_limits<int64_t>::max();
+    std::optional<int64_t> oldest_key = std::nullopt;
     for (const auto& spk_man_pair : m_spk_managers) {
-        oldestKey = std::min(oldestKey, spk_man_pair.second->GetOldestKeyPoolTime());
+        if (!oldest_key.has_value()) {
+            oldest_key = spk_man_pair.second->GetOldestKeyPoolTime();
+            continue;
+        }
+        if (!spk_man_pair.second->GetOldestKeyPoolTime().has_value()) {
+            continue;
+        }
+        oldest_key = std::min(oldest_key.value(), spk_man_pair.second->GetOldestKeyPoolTime().value());
```

Comment:
This is a behaviour change: Previously, it would return 0 if any of the m_spk_managers returned 0. With the new semantics, it should prefer std::nullopt over a value.

If this change is intentional, please document why.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/23348#issuecomment-958736346).